core/src/mindustry/entities/Lightning.java

```diff
@@ -43,7 +43,11 @@ private static void createLightningInternal(@Nullable Bullet hitter, int seed, T
         bhit = false;
 
         for(int i = 0; i < length / 2; i++){
-            hitCreate.create(null, team, x, y, 0f, damage, 1f, 1f, hitter);
+            if(hitter != null){
+                if(hitter.type.lightningTypeChance < Mathf.random() || hitter.type.lightningTypeChance >= 1){
```

Comment:
Couldn't this instead be
`if(hitter != null && (hitter.type.lightningTypeChance < Mathf.random() || hitter.type.lightningTypeChance >= 1)){`

Reply:
That could work

Reply:
Also, I would put `hitter.type.lightningTypeChance >= 1` first so that `Mathf.random()` is not called unnecessarily, like this:
`if(hitter != null && (hitter.type.lightningTypeChance >= 1 || hitter.type.lightningTypeChance < Mathf.random())){`

Ah, _micro-optimizations._